server/models/registration/utils.go

```diff
@@ -0,0 +1,57 @@
+package registration
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1alpha2"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1beta1"
+	"github.com/layer5io/meshkit/models/meshmodel/entity"
+	"gopkg.in/yaml.v2"
+)
+
+func unmarshal(byt []byte, out interface{}) error {
+	err := json.Unmarshal(byt, out)
+	if(err != nil){
+		err = yaml.Unmarshal(byt, out)
+		if(err != nil){
+			return fmt.Errorf("Not a valid YAML or JSON")
```

Comment:
Please replace with meshkit error.

Reply:
this was just meant as a helper function.. the only usage of this function wraps it in a meshkit error.. but yes, I will use meshkit error once moved over to meshkit, as it will be residing in the `utils` package